workloads/imagenet/imagenet_jax/workload.py

```diff
@@ -0,0 +1,290 @@
+      update_batch_norm=False)
+    return self.compute_metrics(logits, batch['label'])
+
+  def model_fn(
+      self,
+      params: spec.ParameterTree,
+      augmented_and_preprocessed_input_batch: spec.Tensor,
+      model_state: spec.ModelAuxillaryState,
+      mode: spec.ForwardPassMode,
+      rng: spec.RandomState,
+      update_batch_norm: bool) -> Tuple[spec.Tensor, spec.ModelAuxillaryState]:
+    variables = {'params': params, **model_state}
+    train = mode == spec.ForwardPassMode.TRAIN
+    if update_batch_norm:
+      logits, new_model_state = self._model.apply(
+        variables,
+        jax.numpy.squeeze(augmented_and_preprocessed_input_batch['image']),
+        train=train,
+        mutable=['batch_stats'])
+      return logits, new_model_state
+    else:
+      logits = self._model.apply(
+        variables,
+        jax.numpy.squeeze(augmented_and_preprocessed_input_batch['image']),
+        train=train,
+        mutable=False)
+      return logits, None
+
+  # Does NOT apply regularization, which is left to the submitter to do in
+  # `update_params`.
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,
+      logits_batch: spec.Tensor) -> spec.Tensor:  # differentiable
+    """Cross Entropy Loss"""
+    one_hot_labels = common_utils.onehot(label_batch,
+                                         num_classes=self.num_classes)
+    xentropy = optax.softmax_cross_entropy(logits=logits_batch,
+                                           labels=one_hot_labels)
+    return jnp.mean(xentropy)
+
+  def compute_metrics(self, logits, labels):
+    loss = self.loss_fn(labels, logits)
+    accuracy = jnp.mean(jnp.argmax(logits, -1) == labels)
+    metrics = {
+      'loss': loss,
+      'accuracy': accuracy,
+    }
+    metrics = lax.pmean(metrics, axis_name='batch')
+    return metrics
+
+  def eval_model(
+      self,
+      params: spec.ParameterTree,
+      model_state: spec.ModelAuxillaryState,
+      rng: spec.RandomState,
+      data_dir: str):
+    """Run a full evaluation of the model."""
+
+    # sync batch statistics across replicas once per epoch
+    model_state = self.sync_batch_stats(model_state)
+
+    epoch_metrics = common_utils.get_metrics(self.epoch_metrics)
+    summary = jax.tree_map(lambda x: x.mean(), epoch_metrics)
+    logging.info('train loss: %.4f, accuracy: %.2f',
+                  summary['loss'], summary['accuracy'] * 100)
+    self.epoch_metrics = []
+    eval_metrics = []
+
+    data_rng, model_rng = jax.random.split(rng, 2)
+    eval_batch_size = self.batch_size
+    num_batches = self.num_eval_examples // eval_batch_size
+    if self._eval_ds is None:
+      self._eval_ds = self._build_dataset(
+        data_rng, split='test', batch_size=eval_batch_size, data_dir=data_dir)
+    eval_iter = iter(self._eval_ds)
+    total_accuracy = 0.
+    eval_step = 1
+    start_time = time.time()
+    accumulated_eval_time = 0
+    for batch in eval_iter:
+      metrics = jax.pmap(self.eval_model_fn,
```

Comment:
instead of having jax.pmap here, can you add it as a function decorator around `eval_model_fn` (similar to what we did with `pmapped_train_step` in `submission.py`)